regcomp.c

```diff
@@ -12763,32 +12763,35 @@ S_regpiece(pTHX_ RExC_state_t *pRExC_state, I32 *flagp, U32 depth)
     }
 
     if ((flags&SIMPLE)) {
-        if (min == 0 && max == REG_INFTY) {
+        if (max == REG_INFTY) {
+            if (min == 1) {
+                reginsert(pRExC_state, PLUS, ret, depth+1);
+                MARK_NAUGHTY(3);
+                goto done_main_op;
```

Comment:
By moving the actual common bits further away from each other, I think this ends up making the commonality _less_ clear. I think it should at least retain the original order of testing `min == 0` first; even then, I'm not at all convinced this improves anything.

What really obscures the commonality is the 17-line chunk for the WILDCARD check. The real benefit to clarity would come from making that a terse check-and-call to a separate function where all the verbose commentary can live; but whether that's worth doing is also debatable.

Reply:
I like the grouping of the * and + cases under a common upper-infinity one to show that commonality.  And that chunk of code I have moved to the end of the function and it is goto'd.  Since it croaks and doesn't return, that's sufficient.  I did reorder the * to be beore the +.  That makes the parallel code readily apparent.